cctrl/auth.py

```diff
@@ -92,12 +132,73 @@ def delete_tokenfile(settings):
     return False
 
 
+def set_configfile(settings, email=None, ssh_auth=None, ssh_path=None):
+    create_config_dir(settings)
+    config = ConfigParser.ConfigParser()
+    config.read(settings.config_path)
+
+    if not config.has_section('user'):
+        config.add_section('user')
+
+    if email:
+        config.set('user', 'email', email)
+
+    if ssh_auth is not None:
+        config.set('user', 'ssh_auth', ssh_auth)
+
+    if ssh_path:
+        config.set('user', 'ssh_path', ssh_path)
+
+    with open(settings.config_path, 'w') as configfile:
+        config.write(configfile)
+
+    return True
+
+
+def get_configfile(settings):
+    config = ConfigParser.ConfigParser()
+    config.read(settings.config_path)
+    if config.has_section('user'):
+        cf = dict(config.items('user'))
+        if 'ssh_auth' in cf:
+            cf['ssh_auth'] = True if cf['ssh_auth'] == "True" else False
+
+        return cf
+
+    return {}
+
+
+def get_email_and_password(settings):
+    email = get_email_env(settings) or \
+        get_configfile(settings).get('email') or \
+        get_email(settings)
+
+    if get_configfile(settings).get('ssh_auth'):
+        return email, None
+
+    password = get_password_env(settings)
+    if not password:
+        password = get_password()
+
+    return email, password
+
+
 def get_email(settings):
     sys.stderr.write(settings.login_name)
     sys.stderr.flush()
 
     email = raw_input()
+    set_configfile(settings, email=email)
```

Comment:
Yes, in general we want to store the email in the config file. e.g. when running `cctrluser setup` for first time without specifying `--email` flag, we ask for the email and store it. I think it's a good approach. Not sure about storing it when getting the email from the environment (see next method), let me know it it makes sense there (e.g. you have a default email stored on the config but for whatever reason you want to run a command with another account without changing default setup).